src/ctap/pin_protocol_v1.rs

```diff
@@ -136,12 +136,31 @@ fn check_and_store_new_pin(
         // TODO(kaczmarczyck) check last byte == 0x00
         return false;
     }
-    let mut pin_hash = [0; 16];
+    let mut pin_hash = [0u8; 16];
     pin_hash.copy_from_slice(&Sha256::hash(&pin[..])[..16]);
     persistent_store.set_pin_hash(&pin_hash);
     true
 }
 
+#[cfg(feature = "with_ctap2_1")]
+// TODO remove when all variants are used
+#[allow(dead_code)]
+pub enum PinPermission {
+    MakeCredential = 0x01,
+    GetAssertion = 0x02,
+    CredentialManagement = 0x04,
+    BioEnrollment = 0x08,
+    PlatformConfiguration = 0x10,
+    AuthenticatorConfiguration = 0x20,
+}
+
+#[cfg(feature = "with_ctap2_1")]
+impl PinPermission {
+    pub fn check(self, stored_bits: u8) -> bool {
```

Comment:
I'm not sure why we can't inline this function. It seems to be used only once. And the name of the function where it is used is much clearer than `check`. Otherwise I would add at least some documentation on the expected semantics here.

Reply:
We can, if we want that. I thought it belongs to permission more than it belongs to the protocol, but I am happy to change that.